src/OrleansProviders/LogViews/SharedStorageProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.LogViews;
+using Orleans.Runtime;
+using Orleans.Runtime.LogViews;
+using Orleans.Storage;
+using System.Threading;
+
+namespace Orleans.Providers.LogViews
+{
+    /// <summary>
+    /// A log view provider that stores the latest view in primary storage, using any standard storage provider.
+    /// Supports multiple clusters connecting to the same primary storage (doing optimistic concurrency control via e-tags)
+    ///<para>
+    /// The log itself is transient, i.e. not actually saved to storage - only the latest view (snapshot) and some 
+    /// metadata (the log position, and write flags) are stored in the primary. 
+    /// </para>
+    /// </summary>
+    public class SharedStorageProvider : ILogViewProvider
+    {
+        public string Name { get; private set; }
+
+        public const string GLOBAL_STORAGE_PROVIDER_PARAMETER = "GlobalStorageProvider";
+
+        public Logger Log { get; private set; }
+
+        private static int counter;
```

Comment:
What does this count?

In general (maybe someone has an idea), I wonder if this would be problematic when running multi-silo cluster on one computer (say, a dev-box in reliable setup) or if it would work together with the DI implementation currently coming along too.